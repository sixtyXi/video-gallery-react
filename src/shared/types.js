// @flow
import type { RecordFactory, RecordOf } from 'immutable';
import { Record } from 'immutable';

type MovieProps = {
  id: string | number,
  poster_path: string,
  title: string,
  genres: Array<string>,
  release_date: string,
  vote_average: number,
  tagline: string,
  runtime: number,
  overview: string
};

const defaultValuesMovie: MovieProps = {
  id: '',
  poster_path: '',
  title: '',
  genres: [],
  release_date: '',
  vote_average: 0,
  tagline: '',
  runtime: 0,
  overview: ''
};

export type Movie = RecordOf<MovieProps>;

export const makeMovie: RecordFactory<Movie> = Record(defaultValuesMovie);

export type Filter = {
  value: string,
  title: string
};

export type MovieFilters = {
  searchBy: string,
  sortBy: string,
  search: string
};

export type Location = {
  search: string,
  pathname: string
};
