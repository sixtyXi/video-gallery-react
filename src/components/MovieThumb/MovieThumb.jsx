import React from 'react';

import styles from './MovieThumb.scss';

const mapGenres = genres => {
  return genres.map(genre => genre.replace(/^./, genre[0].toUpperCase())).join(' & ');
};

export const MovieThumb = ({ movie }) => {
  if (!movie) return null;

  const { genres = [], poster_path: posterPath, title, release_date: releaseDate } = movie;

  return (
    <article className={styles.wrapper}>
      <img className={styles.poster} src={posterPath} alt={`Poster of ${title}`} />
      <div className={styles.titleWrapper}>
        <h2 className={styles.title}>{title}</h2>
        <time className={styles.releaseDate}>{releaseDate}</time>
      </div>
      <p className={styles.genre}>{mapGenres(genres)}</p>
    </article>
  );
};
