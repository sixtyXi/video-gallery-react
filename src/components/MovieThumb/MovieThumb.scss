@import '../../shared/_typography.scss';

.wrapper {
  display: flex;
  flex-wrap: wrap;
  width: 200px;
  font-size: $size-s;
  line-height: 1.2;
  cursor: pointer;
}

.poster {
  width: 100%;
  height: 280px;
}

.titleWrapper {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  padding: 5px;
}

.title {
  font-size: inherit;
  line-height: inherit;
  font-variant: all-small-caps;
}

.releaseDate {
  padding: 0 5px;
  border: 1px solid;
}

.genre {
  width: 100%;
  padding: 0 5px;
  color: gray;
}
