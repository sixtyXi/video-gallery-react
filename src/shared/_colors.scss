$accent: #fd4d61;
$secondary: #414141;
