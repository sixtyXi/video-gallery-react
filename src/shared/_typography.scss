$text-light: white;
$size-m: 14px;
$size-s: 12px;
$size-xs: 10px;
$size-l: 16px;
