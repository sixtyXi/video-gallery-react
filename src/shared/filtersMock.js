export const SORT_FILTERS = [
  {
    value: 'release_date',
    title: 'release date'
  },
  {
    value: 'vote_average',
    title: 'rating'
  }
];

export const SEARCH_FILTERS = [
  {
    value: 'title',
    title: 'title'
  },
  {
    value: 'genres',
    title: 'genre'
  }
];

export const DEFAULT_FILTERS = {
  searchBy: SEARCH_FILTERS[0].value,
  sortBy: SORT_FILTERS[0].value,
  search: ''
};
