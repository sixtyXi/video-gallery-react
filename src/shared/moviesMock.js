export const MOVIES = [
  {
    id: 680,
    title: 'Pulp Fiction',
    tagline: "Just because you are a character doesn't mean you have character.",
    vote_average: 8.3,
    vote_count: 10910,
    release_date: '1994-09-10',
    poster_path: 'https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg',
    overview:
      "A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.",
    budget: 8000000,
    revenue: 213928762,
    genres: ['Thriller', 'Crime'],
    runtime: 154
  },
  {
    id: 2,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 3,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  },
  {
    id: 4,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 5,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  },
  {
    id: 6,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 7,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  }
];
