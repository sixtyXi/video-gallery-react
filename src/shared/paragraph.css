.paragraph {
  height: 50px;
  display: inline-block;
  padding: 10px;
  line-height: 30px;
}
