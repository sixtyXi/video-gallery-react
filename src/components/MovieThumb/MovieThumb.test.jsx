import React from 'react';
import { shallow } from 'enzyme';

import { MOVIES } from '../../shared/moviesMock';
import { MovieThumb } from './MovieThumb';

describe('MovieThumb', () => {
  const movie = MOVIES[1];

  test('renders correctly', () => {
    const wrapper = shallow(<MovieThumb movie={movie} />);

    expect(wrapper).toMatchSnapshot();
  });
});
