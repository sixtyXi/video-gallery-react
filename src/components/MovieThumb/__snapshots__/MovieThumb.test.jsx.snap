// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieThumb renders correctly 1`] = `
<article
  className="wrapper"
>
  <img
    alt="Poster of Kill Bill: Vol. 1"
    className="poster"
    src="https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg"
  />
  <div
    className="titleWrapper"
  >
    <h2
      className="title"
    >
      Kill Bill: Vol. 1
    </h2>
    <time
      className="releaseDate"
    >
      2003
    </time>
  </div>
  <p
    className="genre"
  >
    Action & Adventure
  </p>
</article>
`;
